hack/installers/install-kustomize.sh

```diff
@@ -13,32 +23,32 @@ KUSTOMIZE_VERSION=${KUSTOMIZE_VERSION:-$kustomize4_version}
 # v3.3.0 = https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v3.3.0/kustomize_v3.3.0_linux_amd64.tar.gz
 case $ARCHITECTURE in
   arm|arm64)
-      export TARGET_FILE=kustomize_${KUSTOMIZE_VERSION}_linux_${ARCHITECTURE}.tar.gz
-      URL=https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v${KUSTOMIZE_VERSION}/kustomize_v${KUSTOMIZE_VERSION}_linux_$ARCHITECTURE.tar.gz
+      export TARGET_FILE=kustomize_${KUSTOMIZE_VERSION}_${INSTALL_OS}_${ARCHITECTURE}.tar.gz
+      URL=https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v${KUSTOMIZE_VERSION}/kustomize_v${KUSTOMIZE_VERSION}_${INSTALL_OS}_$ARCHITECTURE.tar.gz
       BINNAME=kustomize
       [ -e ${DOWNLOADS}/${TARGET_FILE} ] || curl -sLf --retry 3 -o ${DOWNLOADS}/${TARGET_FILE} "$URL"
       $(dirname $0)/compare-chksum.sh
       tar -C /tmp -xf ${DOWNLOADS}/${TARGET_FILE}
-      sudo install -m 0755 /tmp/kustomize $BIN/$BINNAME
+      sudo install -m 0755 /tmp/kustomize $DIST_PATH/$BINNAME
```

Comment:
`$PROJECT_ROOT/dist` is the place where we install all Go utility tools binaries. The idea is that we install in this folder all tools with the required/appropriate versions that the project expects and it would avoid issues with devs and CI providing a reproducible build on every environment.